app/controllers/textings_controller.rb

```diff
@@ -5,24 +5,24 @@
 
 class TextingsController < ApplicationController
   def create
-    recipient_name = texting_params[:recipient_name]
-    phone_number = parse_phone_number
-    service_id = texting_params[:service_id]
-    data = get_db_data(recipient_name, phone_number, service_id)
-    # Make a request to Textellent API. If request successful we update our DB
+    text_data = aggregate_text_data
 
-    response = JSON.parse(post_textellent(data).body)
+    # Make a request to Textellent API. If the request is successful, we store a record in our DB
+    response = JSON.parse(post_textellent(text_data).body)
     Rails.logger.info(response)
+
     if response['status'] != 'success'
       render status: :bad_request, json: { error: 'failure' }
       return
     end
-    update_db(recipient_name, phone_number, service_id)
+    update_db
   end
 
   private
 
-  def update_db(recipient_name, phone_number, service_id)
+  def update_db
+    phone_number = texting_params[:phone_number]
+    recipient_name = texting_params[:recipient_name]
```

Comment:
I feel like it would be better to source this information from the already-processed `text_data` variable than to re-read these from the HTTP parameters. This would avoid throwing out any data sanitization that we're doing, such as what the `#parse_phone_number` method does.

Reply:
Good thinking, I've updated it!